cmake/toolchains/mt.runtime.win32.toolchain.cmake

```diff
@@ -27,6 +27,11 @@ if(use_static_runtime)
         foreach(build_type "" "_DEBUG" "_MINSIZEREL" "_RELEASE" "_RELWITHDEBINFO")
             set(flag_var "CMAKE_${lang}_FLAGS${build_type}_INIT")
             string(REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+            if (${build_type} STREQUAL "_DEBUG")
+                set(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} /MTd")'
+            else()
+                set(CMAKE_CXX_FLAGS_INIT "${CMAKE_CXX_FLAGS_INIT} /MT")'
```

Comment:
Looks like we need to populate ${flag_var} here

BTW, I wonder whether RELWTIHDEBINFO should also use /MTd ?

Reply:
"MultiThreaded$<$<CONFIG:Debug>:Debug>" (ref: https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html), According to this generator expression, I think only for DEBUG, it will be MTd

Reply:
Append `/MT` or `/MTd` to `"CMAKE_${lang}_FLAGS${build_type}_INIT"` will not solve the mismatch issue (finally it will be overriden with '/MD'):
```cl : Command line warning D9025 : overriding '/MT' with '/MD'
[2/11] Building CXX object thirdparty\gflags\gflags\CMakeFiles\gflags_nothreads_static.dir\Release\src\windows_port.cc.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[3/11] Building CXX object thirdparty\gflags\gflags\CMakeFiles\gflags_nothreads_static.dir\Release\src\gflags_reporting.cc.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[4/11] Building CXX object thirdparty\gflags\gflags\CMakeFiles\gflags_nothreads_static.dir\Release\src\gflags_completions.cc.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[5/11] Building CXX object thirdparty\gflags\gflags\CMakeFiles\gflags_nothreads_static.dir\Release\src\gflags.cc.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[6/11] Building CXX object common\utils\CMakeFiles\ie_samples_utils.dir\Release\src\latency_metrics.cpp.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[7/11] Building CXX object hello_query_device\CMakeFiles\hello_query_device.dir\Release\main.cpp.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[8/11] Linking CXX static library intel64\Release\gflags_nothreads_static.lib
[9/11] Building CXX object common\utils\CMakeFiles\ie_samples_utils.dir\Release\src\args_helper.cpp.obj
cl : Command line warning D9025 : overriding '/MT' with '/MD'
cl : Command line warning D9025 : overriding '/MD' with '/MT'
cl : Command line warning D9025 : overriding '/MT' with '/MD'
[10/11] Linking CXX static library intel64\Release\ie_samples_utils.lib
[11/11] Linking CXX executable intel64\Release\hello_query_device.exe
FAILED: intel64/Release/hello_query_device.exe 
C:\Windows\system32\cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=hello_query_device\CMakeFiles\hello_query_device.dir\Release --rc="C:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\x64\rc.exe" --mt="C:\Program Files (x86)\Windows Kits\10\bin\10.0.22000.0\x64\mt.exe" --manifests  -- C:\BuildTools\VC\Tools\MSVC\14.40.33807\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\hello_query_device.Release.rsp  /out:intel64\Release\hello_query_device.exe /implib:intel64\Release\hello_query_device.lib /pdb:intel64\Release\hello_query_device.pdb /version:0.0 /NODEFAULTLIB:combase.lib /NODEFAULTLIB:mscoree.lib /NODEFAULTLIB:ole32.lib /NODEFAULTLIB:advapi32.lib /NODEFAULTLIB:user32.lib /NODEFAULTLIB:kernel32.lib  /NODEFAULTLIB:combase.lib /NODEFAULTLIB:mscoree.lib /NODEFAULTLIB:ole32.lib /NODEFAULTLIB:advapi32.lib /NODEFAULTLIB:user32.lib /NODEFAULTLIB:kernel32.lib  /machine:x64 /LARGEADDRESSAWARE /INCREMENTAL:NO /subsystem:console  /LIBPATH:\"$(VC_LibraryPath_VC_x64_OneCore)\" && cd
LINK: command "C:\BuildTools\VC\Tools\MSVC\14.40.33807\bin\Hostx64\x64\link.exe /nologo @CMakeFiles\hello_query_device.Release.rsp /out:intel64\Release\hello_query_device.exe /implib:intel64\Release\hello_query_device.lib /pdb:intel64\Release\hello_query_device.pdb /version:0.0 /NODEFAULTLIB:combase.lib /NODEFAULTLIB:mscoree.lib /NODEFAULTLIB:ole32.lib /NODEFAULTLIB:advapi32.lib /NODEFAULTLIB:user32.lib /NODEFAULTLIB:kernel32.lib /NODEFAULTLIB:combase.lib /NODEFAULTLIB:mscoree.lib /NODEFAULTLIB:ole32.lib /NODEFAULTLIB:advapi32.lib /NODEFAULTLIB:user32.lib /NODEFAULTLIB:kernel32.lib /machine:x64 /LARGEADDRESSAWARE /INCREMENTAL:NO /subsystem:console /LIBPATH:"$(VC_LibraryPath_VC_x64_OneCore)" /MANIFEST:EMBED,ID=1" failed (exit code 1169) with the following output:
openvino.lib(type.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(except.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(any.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(version.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(core.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(extension.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(remote_context.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj
openvino.lib(compiled_model.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in main.cpp.obj

```